src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -232,6 +217,28 @@ void HSHomeObject::init_homestore() {
     }
 }
 
+void HSHomeObject::on_replica_restart() {
+    std::call_once(replica_restart_flag_, [this]() {
```

Comment:
according to the logic here, when the first repl_dev is recovered , all the pg and shard will be recovered also.
however, pg2 , which depends on repl_dev2, can not be recovered since when repl_dev1 is recoverd, repl_dev2 is not recovered, right?

Reply:
can you explain why pg2 cannot recovered?  the callback get called after all the repl_dev metablk recovered.